backend/handlers/math/outcomes/dbpm/marketshares.go

```diff
@@ -163,42 +147,56 @@ func CalculateScaledPayoutsDBPM(allBetsOnMarket []models.Bet, coursePayouts []Co
 	return scaledPayouts
 }
 
-// adjust payouts to account for case where calculated payouts > available
-func AdjustPayoutsFromNewest(bets []models.Bet, scaledPayouts []int64) []int64 {
-	// Calculate the sum of scaledPayouts
+// Calculate the excess amount based on available pool and scaled payouts
+func calculateExcess(bets []models.Bet, scaledPayouts []int64) int64 {
 	var sumScaledPayouts int64
 	for _, payout := range scaledPayouts {
 		sumScaledPayouts += payout
 	}
-
 	availablePool := marketmath.GetMarketVolume(bets)
+	return sumScaledPayouts - availablePool
+}
 
-	// Determine the excess amount
-	excess := sumScaledPayouts - availablePool
-
-	// Loop to deduct from newest to oldest until there's no excess
+// Adjust scaled payouts if excess is greater than 0
+// This  should not be possible given how the preceeding pipeline works, but we adjust for it anyway.
```

Comment:
I'm worried that I'm not accounting for some odd edge case, game set up or otherwise that would result in a negative excess, which would eliminate money from the system in a way we're not accounting for.  Basically this is a safeguard.